src/main/scala/org/zalando/spearheads/innkeeper/api/JsonProtocols.scala

```diff
@@ -1,122 +1,164 @@
+    override def read(json: JsValue): StrictPathMatcher = ???
+  }
 
-    override def read(json: JsValue): Endpoint = {
-      val endpoint = endpointFormat.read(json)
+  implicit object PathMatcherFormat extends RootJsonFormat[PathMatcher] {
+    def write(pathMatcher: PathMatcher) = pathMatcher match {
+      case RegexPathMatcher(matcher) =>
+        JsObject(Map("match" -> JsString(matcher),
+          "type" -> JsString(MatcherType.REGEX)))
+      case StrictPathMatcher(matcher) =>
+        JsObject(Map("match" -> JsString(matcher),
+          "type" -> JsString(MatcherType.STRICT)))
+    }
 
-      endpoint.copy(
-        port = endpoint.port.orElse(Some(443)),
-        protocol = endpoint.protocol.orElse(Some(Https)),
-        endpointType = endpoint.endpointType.orElse(Some(ReverseProxy))
-      )
+    def read(value: JsValue) = value.asJsObject.getFields("match", "type") match {
+      case Seq(JsString(matcher), JsString(MatcherType.REGEX)) =>
+        new RegexPathMatcher(matcher)
+      case Seq(JsString(matcher), JsString(MatcherType.STRICT)) =>
+        new StrictPathMatcher(matcher)
+      case _ => deserializationError("PathMatcher expected")
     }
   }
 
-  implicit object MatcherTypeFormat extends JsonFormat[MatcherType] {
-    val REGEX = "REGEX"
-    val STRICT = "STRICT"
+  private val matcherFormat = jsonFormat(Matcher, "hostMatcher", "pathMatcher", "methodMatcher", "headerMatchers")
 
-    override def write(matcherType: MatcherType): JsValue = matcherType match {
-      case Regex  => JsString(REGEX)
-      case Strict => JsString(STRICT)
+  implicit object MatcherFormat extends RootJsonFormat[Matcher] {
+
+    private def validate(matcher: Matcher, ex: Exception) = {
+      if ((!matcher.headerMatchers.isDefined || matcher.headerMatchers.get.isEmpty) &&
+        !matcher.hostMatcher.isDefined &&
+        !matcher.methodMatcher.isDefined &&
+        !matcher.pathMatcher.isDefined) {
+        throw ex
+      }
+      matcher
     }
 
-    override def read(json: JsValue): MatcherType = json match {
-      case JsString(REGEX)  => Regex
-      case JsString(STRICT) => Strict
-      case _                => deserializationError("MatcherType expected")
+    override def write(matcher: Matcher): JsValue = {
+      matcherFormat.write {
+        validate(
+          matcher,
+          new SerializationException("At least one of the fields in the matcher should be defined")
+        )
+      }
     }
-  }
 
-  implicit val pathMatcherFormat = jsonFormat(PathMatcher.apply, "match", "type")
+    override def read(json: JsValue): Matcher = {
+      val matcher = matcherFormat.read(json)
 
-  val newRouteFormat = jsonFormat(NewRoute.apply, "description", "match_path", "endpoint",
-    "match_headers", "match_methods", "request_headers", "response_headers", "path_rewrite")
+      val matcherWithDefaults = matcher.copy(
+        hostMatcher = matcher.hostMatcher.orElse(None),
```

Comment:
is `orElse(None)` really needed? Which case is it supposed to handle? The case, where `hostMatcher` is None? Just pass it along as it is...